### PR TITLE
ENH: spectral_density equivalency now handles integrated flux

### LINKED ISSUE #7593: Equivalency for flux
Currently we have flux density but there's none that I can find to convert, say, ph/cm^2/s to erg/cm^2/s. Did I miss something? If I were to implement it, is it okay to stick it in spectral_density or should it go somewhere new? What should I name the new one? spectral_nondensity? spectral_flux? spectral_integrated_flux?

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add handling of integrated flux units to `spectral_density` equivalency.

Fixes #7593

### CODE REVIEW

File: astropy/units/tests/test_equivalencies.py [hunk 1]
```diff
@@ -307,6 +307,16 @@ def test_spectraldensity2():
...
 
+    # integrated flux
+    f_int = u.erg / u.cm ** 2 / u.s
+    phot_int = u.ph / u.cm ** 2 / u.s
+
+    a = f_int.to(phot_int, 1, u.spectral_density(u.Quantity(3500, u.AA)))
+    assert_allclose(a, 1.7619408e+11)
+
+    a = phot_int.to(f_int, 1, u.spectral_density(u.Quantity(3500, u.AA)))
+    assert_allclose(a, 5.67555959e-12)
```
Comment (mhvk) [inline astropy/units/tests/test_equivalencies.py]: Could you add tests where `wav` is a frequency or energy?

Comment (pllim) [reply astropy/units/tests/test_equivalencies.py]: Done.

**Comment by pllim** [comment]
@mhvk , I updated the tests as you recommended. Please have another look. Thank you!

File: astropy/units/tests/test_equivalencies.py [hunk 2]
```diff
@@ -477,6 +491,24 @@ def test_spectraldensity6():
...
 
 
+@pytest.mark.parametrize(
+    ('from_unit', 'to_unit'),
+    [(u.ph / u.cm ** 2 / u.s, (u.cm * u.cm * u.s) ** -1),
+     (u.ph / u.cm ** 2 / u.s, u.erg / (u.cm * u.cm * u.s * u.keV)),
+     (u.erg / u.cm ** 2 / u.s, (u.cm * u.cm * u.s) ** -1),
+     (u.erg / u.cm ** 2 / u.s, u.erg / (u.cm * u.cm * u.s * u.keV))])
+def test_spectraldensity_not_allowed(from_unit, to_unit):
+    """Now allowed to succeed as
```
Comment (mhvk) [inline astropy/units/tests/test_equivalencies.py]: Now->Not

Comment (pllim) [reply astropy/units/tests/test_equivalencies.py]: Ops! Good catch. That does change the meaning completely. 😅

**Comment by pllim** [comment]
p.s. `mpldev` failure is unrelated.

**Comment by pllim** [comment]
I don't understand why Travis CI is failing. The errors do not make sense to me.

**Comment by bsipocz** [comment]
Merging as CI failure is unrelated and has been fixed in master already.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**